### DISCUSSION TITLE
Enabling invite only feature on Sunbird ED

### DISCUSSION THREAD

**Discussion by Xitija** [Sunbird-Ed/Community#341] [Sunbird ED configuration discussion]
Hello All ,

I want to enable the invite only feature for courses on Sunbird Ed on the frontend.
I do not have much information about this feature.
If possible can anyone share what APIs are needed and how frontend changes are implemented?

Please guide me here.

cc:  @coolbung @vinod303 @Renuka-S @komalm @@Krishnaj20

**Comment by Krishnaj20** [comment]
@Xitija - I believe this is a duplicate of https://github.com/orgs/Sunbird-Ed/discussions/274 created by Parth. Can we close this thread and track all discussions in the other thread. 
Re: this capability, we are seeking contributions from the community to enable this capability. The same is updated in the ED roadmap [here](https://project-sunbird.atlassian.net/wiki/spaces/SUN/pages/3259596873/ED+Roadmap). 

@vinukumar-vs @rhwarrier

**Reply by Xitija** [reply]
Hello Sir,

I dont think the issue you are referring to is same as this one.
Here I am asking about a feature that already exists in SunbirdEd and not related to privacy.
It is about creating an invite only batch where only specfic users are added to the course and only they can access it.

**Reply by Krishnaj20** [reply]
@Xitija - invite only is also covered as one of the batch types in the discussion thread.

**Comment by Renuka-S** [comment]
@Krishnaj20 Yes eventually this could be referred to as the above ticket.

For this discussion, we are looking into the specific feature "invite-only" which is available in sunbird. Refering to the discussion https://github.com/Sunbird-Lern/Community/discussions/14 here, backend PR details are provided although, for the frontend code PR, we have started this email thread. 

We are on 4.4 currently so we need to pull back this feature which was later enabled back in 4.9

cc: @Xitija @coolbung @vinod303 @komalm

**Reply by Krishnaj20** [reply]
@Renuka-S - to my knowledge the front end code has been deprecated for invite-only feature. 

@rajeevsathish @vinukumar-vs - Pls share details about this.

**Reply by Krishnaj20** [reply]
@rajeevsathish @vinukumar-vs - can you respond to this thread?

**Comment by rajeevsathish** [comment]
@Renuka-S please refer this html and respective ts file we had invite-only batches till 2.10.3 release 
https://github.com/Sunbird-Ed/SunbirdEd-portal/blob/release-2.10.3/src/app/client/src/app/modules/learn/components/batch/create-batch/create-batch.component.html

cc: @Krishnaj20 @vinukumar-vs

**Reply by Krishnaj20** [reply]
thanks @rajeevsathish . 

@Renuka-S - You can use the sample reference front end code (which is currently deprecated) shared by Rajeev to enable the feature since you've already mentioned that you are aware of the backend details. Request you to share this back as a contribution.

**Reply by Renuka-S** [reply]
Thank you @Krishnaj20. We are looking into this. 

@Xitija @komalm Please check and confirm. @Krishnaj20 We might have a few queries with the current structure so will keep this thread open for a couple of days while we implement it on our side.

**Reply by chitranshu-keshav** [reply]
@Renuka-S Can we mark this thread as answered now?
cc: @Krishnaj20
